internal/databases/mysql/backup_fetch_handler.go

```diff
@@ -12,7 +12,19 @@ func HandleBackupFetch(folder storage.Folder,
 	targetBackupSelector internal.BackupSelector,
 	restoreCmd *exec.Cmd,
 	prepareCmd *exec.Cmd) {
-	internal.HandleBackupFetch(folder, targetBackupSelector, internal.GetCommandStreamFetcher(restoreCmd))
+
+	backup := internal.GetBackup(folder, targetBackupSelector)
+	var sentinel StreamSentinelDto
+	err := backup.FetchSentinel(&sentinel)
+	if err != nil {
+		tracelog.ErrorLogger.FatalfOnError("Fail to fetch backup sentinel: %v", err)
+	}
+	if sentinel.FileNames == nil || len(sentinel.FileNames) == 0 {
```

Comment:
You can leave just the second expression since the length of the nil slice is zero.